src/DomainLib.Persistence.EventStore/EventStoreEventsRepository.cs

```diff
@@ -57,107 +56,76 @@ public async Task<long> SaveEventsAsync<TEvent>(string streamName, long expected
                 foreach (var eventData in eventDatas)
                 {
                     Log.LogTrace("Event to append {EventId}. EventType {EventType}. WriteId {WriteId}. " +
-                                     "EventBytes {EventBytes}. MetadataBytes {MetadataBytes}. IsJson {IsJson} ",
-                                     eventData.EventId, eventData.Type, writeId, eventData.Data, eventData.Metadata, eventData.IsJson);
+                                     "EventBytes {EventBytes}. MetadataBytes {MetadataBytes}. ContentType {ContentType} ",
+                                     eventData.EventId, eventData.Type, writeId, eventData.Data, eventData.Metadata, eventData.ContentType);
                 }
             }
 
-            WriteResult writeResult;
+            IWriteResult writeResult;
             try
             {
-                if (eventDatas.Length <= MaxWriteBatchSize)
-                {
-                    writeResult = await _connection.AppendToStreamAsync(streamName, streamVersion, eventDatas);
-                    Log.LogDebug("Written events to stream. WriteId {WriteId}", writeId);
-                }
-                else
-                {
-                    using var transaction = await _connection.StartTransactionAsync(streamName, streamVersion);
-                    Log.LogDebug("Writing events as transaction. Transaction Id {TransactionId}. Write Id {WriteId}", transaction.TransactionId, writeId);
-                    var sliceStartPosition = 0;
-                    var eventsToSave = true;
-
-                    while (eventsToSave)
-                    {
-                        var eventsSlice = eventDatas.Skip(sliceStartPosition).Take(MaxWriteBatchSize);
-                        await transaction.WriteAsync(eventsSlice);
-                        Log.LogDebug("Slice written to transaction {TransactionId}. Write Id {WriteId}", transaction.TransactionId, writeId);
-
-                        sliceStartPosition += MaxWriteBatchSize;
-
-                        if (sliceStartPosition >= eventDatas.Length)
-                        {
-                            eventsToSave = false;
-                        }
-                    }
-
-                    writeResult = await transaction.CommitAsync();
-                    Log.LogDebug("Transaction {TransactionId} committed. Write Id {WriteId}", transaction.TransactionId, writeId);
-                }
+                writeResult = await _client.AppendToStreamAsync(streamName, streamVersion, eventDatas);
```

Comment:
This seems to have been extremely reduced and a lot more simple now?

I'm assuming all the batching/transactional stuff is either not needed or done internally by the _client?

Reply:
Yep, it exposes an IAsyncEnumerable now and does everything through that. No batching on our side.
One upshot of that is that loading lots of events is significantly slower. I couldn't find a better way to do that.
Might raise an issue with EventStore?